Email service

na106 opened the pull request:
Installed brevo_csharp package in Visual Studio.
Added appsettings.json into .gitignore.
Refactored repository, interface, controller, and program.cs for sending test email.

na106 committed:
refactored to configure email api key from appsettings

na106 committed:
refactored to configure email api key from appsettings

jjcanell77 commented on dotnet/LADP_EFC/Repository/RepositoryEmail.cs:
```diff
@@ -0,0 +1,77 @@
+
+namespace LADP_EFC.Repository
+{
+    public class RepositoryEmail : IRepositoryEmail
+    {
+
+        private readonly string baseUrl;
+        private BrevoApi brevo;
+
+        public RepositoryEmail(IOptions<AppSettings> config, IOptions<BrevoApi> brevoApi)
+        {
+            brevo = brevoApi.Value;
+            baseUrl = config.Value.BaseUrl;
+        }
```
Should use primary constructor like:

```
  public RepositoryEmail(IOptions<AppSettings> config, IOptions<BrevoApi> brevoApi) : IRepositoryEmail
    {

        private readonly string baseUrl = brevoApi.Value;
        privater eadonly  BrevoApi brevo = config.Value.BaseUrl;
```

jjcanell77 commented on dotnet/LADP_EFC/appsettings.json:
```diff
@@ -8,5 +8,13 @@
+    "SenderEmail": "Your Brevo Email",
+    "SenderName": "LADP"
+  },
+  "AppSettings": {
```
Why don't you just combine these?

na106 replied:
This was added for future functions, but can be incorporated together in the next push.

jjcanell77 replied:
It does not show that you are using the "AppSettings": {"BaseUrl": "http://localhost:5173" }
So I would recommend removing.

na106 committed:
Appsettings.json

Removed Appsettings value that included url intended for future use

jjcanell77 commented on dotnet/LADP_EFC/Program.cs:
```diff
@@ -24,7 +27,8 @@ public static void Main(string[] args)
             });
-
+            builder.Services.Configure<AppSettings>(
+                builder.Configuration.GetSection("AppSettings"));
```
You removed the json object but not this which is is also no longer being used.

jjcanell77 commented on dotnet/LADP_EFC/Program.cs:
```diff
@@ -4,6 +4,8 @@
 using LADP_EFC.Repository;
 using LADP_EFC.Repository.Interfaces;
 using System.Text.Json.Serialization;
+using LADP_EFC.Models.AppSettings;
```
This also can be removed.

jjcanell77 commented on dotnet/LADP_EFC/Models/Appsettings/Appsettings.cs:
```diff
@@ -0,0 +1,7 @@
+{
+    public class AppSettings
+    {
+        public string BaseUrl { get; set; }
```
And this as well.

jjcanell77 reviewed (changes requested):
Should check to add any unused code from PRs.

na106 committed:
Update Program.cs

Removed unnecessary code regarding appsettings model

na106 committed:
Delete dotnet/LADP_EFC/Models/Appsettings/Appsettings.cs

Deleted Appsettings.cs

jmedina1329 commented on dotnet/LADP_EFC/Repository/RepositoryEmail.cs:
```diff
@@ -0,0 +1,77 @@
+                    },
+                    To = new List<SendSmtpEmailTo>
+                    {
+                        new SendSmtpEmailTo("na106@mail.chapman.edu")
```
Remove hard-coded email

jjcanell77 replied:
Still shows hard-coded email. You should make it so it grabs it from the appsettings but of course do not use the actual email. Recommend adding your version of filled out appsetting to the trello page for us to have

na106 committed:
Update RepositoryEmail.cs

Removed hard-coded parts

na106 committed:
Update RepositoryEmail.cs

Refactored to use email address value from appsettings.json for TestEmail()

na106 committed:
Update BrevoApi.cs

Added TestRecipient for TestEmail() to utilize email address value from appsettings.json

na106 committed:
Update appsettings.json

Added TestRecipient to BrevoApi for TestEmail() email address values
